Lib/distutils/tests/test_dist.py

```diff
@@ -354,8 +354,11 @@ def test_classifier_invalid_type(self):
         attrs = {'name': 'Boa', 'version': '3.0',
                  'classifiers': ('Programming Language :: Python :: 3',)}
         msg = "'classifiers' should be a 'list', not 'tuple'"
-        with self.assertRaises(TypeError, msg=msg):
-            Distribution(attrs)
+        with self.assertWarns(RuntimeWarning) as warns:
+            d = Distribution(attrs)
+            self.assertIsInstance(d.metadata.classifiers, list)
```

Comment:
Would be better to move these checks outside of the `with` block. Otherwise it isn't clear what operation raises a RuntimeWarning.